src/logic/hook.rkt

```diff
@@ -90,49 +86,58 @@
   )
 
 (define-metafunction formality-logic
-  ;; The "grammar" for predicates is just *any term* -- that's not very
-  ;; useful, and extension languages refine it. When matching against predicates,
-  ;; then, we can use this function to avoid matching on other kinds of goals.
-  is-predicate? : Env Goal -> boolean
+  ;; Categories the `Goal` according to the grammar defined by the upper layers
```

Comment:
"Upper" as in `formality-ty` and `formality-decl`? Is there a more descriptive way to phrase this that helps the reader understand the relationship between layers?

Reply:
I guess higher-level is somewhat more clear